lib/check-meta.nix

```diff
@@ -0,0 +1,200 @@
+let
+  lib = import ./default.nix;
+in rec {
+
+  # Extend a derivation with checks for brokenness, license, etc.
+  # Throw a descriptive error when the check fails; return derivaitonArg otherwise.
+  # Note: no dependencies are checked in this step.
+  addMetaChecks = { config, platform, meta, derivationArg, mkDerivationArg }: let
```

Comment:
I looked at it from the other side &ndash; called it `platform` because of being matched to `meta.platforms`, but I don't have any strong opinion about it.